http4k-format/core/src/test/kotlin/org/http4k/format/AutoContentNegotiatorTest.kt

```diff
@@ -0,0 +1,113 @@
+package org.http4k.format
+
+import com.natpryce.hamkrest.assertion.assertThat
+import org.http4k.core.Body
+import org.http4k.core.ContentType
+import org.http4k.core.HttpHandler
+import org.http4k.core.Method.GET
+import org.http4k.core.Request
+import org.http4k.core.Response
+import org.http4k.core.Status.Companion.OK
+import org.http4k.core.Status.Companion.BAD_REQUEST
+import org.http4k.core.with
+import org.http4k.hamkrest.hasBody
+import org.http4k.hamkrest.hasStatus
+import org.http4k.lens.ContentNegotiation
+import org.http4k.lens.Header
+import org.http4k.lens.string
+import org.junit.jupiter.api.Test
+
+class AutoContentNegotiatorTest {
+
+    private val v1Lens = Body.string(ContentType("v1"))
+        .map({ it.replace("v1-", "") }, { "v1-$it" })
+        .toLens()
+
+    private val v2Lens = Body.string(ContentType("v2"))
+        .map({ it.replace("v2-", "") }, { "v2-$it" })
+        .toLens()
+
+    private val negotiator = ContentNegotiation.auto(v1Lens, v2Lens)
+
+    private val returnJohn: HttpHandler = {
+        Response(OK).with(negotiator.outbound(it) of "john")
+    }
+
+    private val receiveJohn: HttpHandler = { req ->
+        val content = negotiator(req)
+        Response(if (content == "john") OK else BAD_REQUEST)
+    }
+
+    @Test
+    fun `response body - without accept header`() {
+        val response = Request(GET, "/")
+            .let(returnJohn)
+
+        assertThat(response, hasBody("v1-john"))
+    }
+
+    @Test
+    fun `response body - with v1 accept header`() {
+        val response = Request(GET, "/")
+            .header("accept", v1Lens.contentType.toHeaderValue())
+            .let(returnJohn)
+
+        assertThat(response, hasBody("v1-john"))
+    }
+
+    @Test
+    fun `response body - with v2 accept header`() {
+        val response = Request(GET, "/")
+            .header("accept", v2Lens.contentType.toHeaderValue())
```

Comment:
Can the `ACCEPT` lens be used to inject a `Content-Type` into the request?  Seems like it's only outbound.

Reply:
it is only an extractor at the moment - mostly because we didn't want to get into dealing with the complexity of the directives and q-levels: ```Accept: text/html, application/xhtml+xml, application/xml;q=0.9, image/webp, */*;q=0.8```